src/DynamicProbes.csproj

```diff
@@ -21,4 +20,35 @@
     <Compile Remove="$(GeneratedFolder)/**/*.cs" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Include="Microsoft.CodeAnalysis.PublicApiAnalyzers" Version="3.3.4">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
+
+  <ItemGroup>
+    <None Update="Probes.g.tt">
+      <Generator>TextTemplatingFileGenerator</Generator>
+      <LastGenOutput>Probes.g.cs</LastGenOutput>
+    </None>
+  </ItemGroup>
+
+  <ItemGroup>
+    <Service Include="{508349b6-6b84-4df5-91f0-309beebad82d}" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Compile Update="Probes.g.cs">
+      <DesignTime>True</DesignTime>
+      <AutoGen>True</AutoGen>
+      <DependentUpon>Probes.g.tt</DependentUpon>
+    </Compile>
+  </ItemGroup>
```

Comment:
If I understand correctly, this is metainformation about the sources for the build task, but it won't regenerate if the template changes. At least on my Ubuntu, `dotnet build` doesn't trigger `Probes.g.cs` regeneration.

One could define a pre-build task [to do exactly that](https://github.com/nogic1008/T4Sample/tree/master?tab=readme-ov-file#define-prebuild-task). But it will be noisy if we keep the `generated at {datetime}...` line in the generated file.

CI can still check if the generated source is in sync with the template by making an exception for `generated at {datetime}` in the diff.

What do you think? Is there a best practice for keeping T4 templates in sync with the generated files?